WebDriverAgentLib/Utilities/FBXCAXClientProxy.m

```diff
@@ -73,13 +40,11 @@ - (BOOL)setAXTimeout:(NSTimeInterval)timeout error:(NSError **)error
                                      maxDepth:(nullable NSNumber *)maxDepth
                                         error:(NSError **)error
 {
-  NSMutableDictionary *parameters = [[NSMutableDictionary alloc] init];
-  // Mimicking XCTest framework behavior (this attribute is added by default unless it is an excludingNonModalElements query)
-  // See https://github.com/appium/WebDriverAgent/pull/523
-  if (SYSTEM_VERSION_GREATER_THAN_OR_EQUAL_TO(@"13.0")) {
-    parameters[@"snapshotKeyHonorModalViews"] = @(NO);
-  }
+  // Setting parameters argument to nil enforces `requestSnapshotForElement` to
+  // fetch default values using the swizzled [XCAXClient_iOS defaultParameters] API call.
```

Comment:
I've ran a test and saw that swizzling the `defaultParameters` attribute prevents the `snapshotKeyHonorModalViews` attribute from being added to the custom `requestSnapshotForElement` call, so I believe this parameter should be added here as it was previously.

![image](https://user-images.githubusercontent.com/34237538/214517991-e02853e9-3386-4955-95dc-86059cefe4fc.png)

Reply:
I simply return what XCTest provides to me in the original API call without adding any extra values. Perhaps, this has been changes since iOS 13. Have you verified v16, for example?

Reply:
The tested device in this case is running iOS 15.7.2, and according to the framework from the latest Xcode the iOS 13 condition is still there and is still relevant for iOS 16:
![image](https://user-images.githubusercontent.com/34237538/214524511-54e39d79-9583-4058-8e35-ebe1d8ac9be2.png)

The thing is, a regular snapshot retrieval path as part of resolving a query goes through `[XCTElementQuery _snapshotForElement:error:]` before calling `requestSnapshotForElement`, and this is the part where the `snapshotKeyHonorModalViews` is added (as part of calling `[XCTElementQuery snapshotParameters]`):
![image](https://user-images.githubusercontent.com/34237538/214524871-c2f1d2ae-c732-4acb-8bc9-a0682e0099d0.png)

Reply:
ok, thanks for the details

Reply:
this is now updated @Dan-Maor 
btw, do you have any sample application to verify whether this patch fixes issues similar to https://github.com/appium/appium/issues/18107 or at least confirm it is an issue at all?

Reply:
Looks good now, thanks.
Regarding https://github.com/appium/appium/issues/18107 - I'll have to check, will do so tomorrow. I suspect though that this is more likely to be related to the target device noted in the issue running iOS 16.3, or perhaps the user's app is now built with a newer Xcode version where the accessibility ID is not exposed correctly. I'll run some tests tomorrow to see if I can replicate the issue.